code/datums/elements/tool_renaming.dm

```diff
@@ -0,0 +1,67 @@
+/**
+ * Renaming tool element
+ *
+ * When using this tool on an object with UNIQUE_RENAME,
+ * lets the user rename/redesc it.
+ */
+/datum/element/tool_renaming
+
+/datum/element/tool_renaming/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ITEM_INTERACTING_WITH_ATOM, PROC_REF(attempt_rename))
+
+/datum/element/tool_renaming/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_INTERACTING_WITH_ATOM)
+
+/datum/element/tool_renaming/proc/attempt_rename(datum/source, mob/living/user, atom/interacting_with, list/modifiers)
+	SIGNAL_HANDLER
+
+	if(!isobj(interacting_with))
+		return NONE
+	
+	var/obj/renamed_obj = interacting_with
+
+	if(!(renamed_obj.obj_flags & UNIQUE_RENAME))
+		return NONE
+
+	var/pen_choice = tgui_input_list(user, "What would you like to edit?", "Pen Setting", list("Rename", "Description", "Reset"))
+	if(QDELETED(renamed_obj) || !user.can_perform_action(renamed_obj))
+		return ITEM_INTERACT_BLOCKING
+
+	switch(pen_choice)
+		if("Rename")
```

Comment:
Can we make these strings into defines? 
I know they're only used twice and we're not that likely to expand the list but retyping the same string when it's used for logic bugs me, just begging for a spelling error to occur one day.

Reply:
hrmn, yeah probably